lib/signaling/v2/peerconnection.js

```diff
@@ -474,10 +483,15 @@ class PeerConnectionV2 extends StateMachine {
   _handleIceConnectionStateChange() {
     const { iceConnectionState } = this._peerConnection;
 
-    this._log.debug(`ICE connection state is "${iceConnectionState}"`);
+    this._log.debug(`makarand: ICE connection state is "${this._lastIceConnectionState}" => "${iceConnectionState}"`);
+    // makarand: TODO: is it okay to short-circuit here? or anybody depends on
+    // iceConnectionStateChanged firing on every non-change
+    if (this._lastIceConnectionState === iceConnectionState) {
```

Comment:
@makarandp0 Let's retain how it was before, unless you can demonstrate that "iceConnectionStateChanged" is emitted even when there is a non-change.

Reply:
Looks like the code previously assumed that event could be emitted for non-changes. Notice that https://github.com/twilio/twilio-video.js/blob/master/lib/signaling/v2/peerconnection.js#L494  
explicitely ensures that old state is not "failed" 
```
    // Case 1: Transition to "failed".
    if (this._lastIceConnectionState !== 'failed' && iceConnectionState === 'failed' && !this._shouldRestartIce && !this._isRestartingIce) {
```

This change allows to get rid of that check.